govcd/vm_test.go

```diff
@@ -880,3 +880,46 @@ func (vcd *TestVCD) Test_VMPowerOnPowerOff(check *C) {
 	check.Assert(err, IsNil)
 	check.Assert(vmStatus, Equals, "POWERED_OFF")
 }
+
+// Test gathering VM virtual hardware items
+func (vcd *TestVCD) Test_GetVirtualHardwareSection(check *C) {
+	itemName := "TestGetVirtualHardwareSection"
+
+	if vcd.skipVappTests {
+		check.Skip("Skipping test because vapp wasn't properly created")
+	}
+
+	fmt.Printf("Running: %s\n", itemName)
+
+	// Find VM
+	vapp := vcd.findFirstVapp()
+	vmType, vmName := vcd.findFirstVm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+	vm, err := vcd.client.Client.FindVMByHREF(vmType.HREF)
+	check.Assert(err, IsNil)
+
+	// Set VirtualHardwareSection of found vm
+	vm.VM.VirtualHardwareSection = &types.VirtualHardwareSection{
+		Item: []*types.VirtualHardwareItem{
+			&types.VirtualHardwareItem{
+				InstanceID:          1,
+				AutomaticAllocation: true,
+				Address:             "1.1.1.1",
+				CoresPerSocket:      2,
+			},
+		},
+	}
+
+	// Preform check of virtual hardware section
+	section, err := vm.GetVirtualHardwareSection()
```

Comment:
Instead of checking  if the VM has `VirtualHardwareSection` as it was preset you could use regex'es to validate if the fields are actually as they should be. For example, if the `Address` fields contains MAC address, so you could check it like that:
```
check.Assert(field_name, Matches, "^[a-fA-F0-9:]{17}|[a-fA-F0-9]{12}$")
```

Reply:
Ahh that is a good idea, I have added that to the latest commit.